documentation/specs/rar-as-service.md

```diff
@@ -0,0 +1,216 @@
+# Resolve Assembly Reference as Service Design
+
+This document describes the Resolve Assembly Reference as Service
+
+# Background
+
+[MSBuild](https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild?view=vs-2019) is a universal build engine, used for building pretty much everything in the Microsoft world. It is available on command line (msbuild, [dotnet build](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-build)), runs under the covers when building projects and solutions in Visual Studio, and is used as the local build engine in "higher-order" distributed build systems. Essentially all .NET applications use MSBuild as their primary build engine.
```

Comment:
Most of the background section was taken from @ladipro document, where he introduced me this project for my internship. I have had link to that (and still is in the docx version, albeit that version is now little outdated), but removed it since that document is internal. 

I don't want to suggest what he thought by that, but how I explain it is, that MSBuild can be used for example with [QuickBuild](https://www.pmease.com/).

Reply:
I have changed the docs links, even though when I get to the docs and they are in my local language I always switch them to English. :)